### ISSUE TITLE
[TOOL] Screensaver Time=Never results errors on subsequent connections

### ISSUE THREAD

**Issue by SzieberthAdam** [x86fr/ATX2AT-Smart-Converter#4]
As title sais. Tool reports FW 1.12, HW display shows 1.13.

Clicking on Factory Reset solves the error. Highest option working is 4 hours.

Beginning of Log (infinite loop):
```
19:57:46: Found ATX2AT Smart Converter [COM4]
19:57:46: Connection Successful
19:57:47: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:47: Buffer Overrun!

19:57:47: Wrong header: 
<
_<
_<
_<
_<
_<
_<
ó19:57:48: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:48: Buffer Overrun!

19:57:48: Wrong header: 
_<
_<
_<
_<
_<
_<
ó19:57:49: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:49: Buffer Overrun!

19:57:49: Wrong header: 
P_<
_<
_<
_<
_<
_<
_<
_<
ó19:57:50: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:50: Buffer Overrun!

19:57:50: Wrong header: 

_<
_<
_<
_<
_<
ó19:57:51: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:51: Buffer Overrun!

19:57:51: Wrong header: 
PPPP_<
_<
_<
_<
_<
_<
_<
_<
ó19:57:52: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:52: Buffer Overrun!

19:57:52: Wrong header: 
_<
_<
_<
_<
_<
_<
_<
ó19:57:53: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:53: Buffer Overrun!

19:57:53: Wrong header: 
----____PPPP_<
_<
_<
_<
_<
_<
_<
_<
ó19:57:54: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:54: Buffer Overrun!

19:57:54: Wrong header: 
€    AAAAnnnn‡‡‡‡----____PPPP_<
_<
_<
_<
_<
_<
_<
_<19:57:54: Wrong header: 

ó19:57:55: Data Receive Failed: System.FormatException: Input string was not in a correct format.
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous)
   at System.Windows.Forms.Control.Invoke(Delegate method, Object[] args)
   at ATX2AT_Configuration_Tool.Main.ProcessRAWData(Byte[] BytesRcvd) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\Main.cs:line 109
   at ATX2AT_Configuration_Tool.SerCOM.ProcessSerialBytes(Object sender, SerialDataReceivedEventArgs e) in D:\Dev\ATX2AT\Conf Tool\ATX2AT Configuration Tool\ATX2AT Configuration Tool\SerCOM.cs:line 181
19:57:55: Buffer Overrun!
```

**Comment by x86fr** [comment]
Probably related to your locale.

What do you use as Windows' primary language?

**Comment by SzieberthAdam** [comment]
Could be. English Win 7, but locale is Hungarian (Hungary).

**Comment by x86fr** [comment]
Can you check if the setting that cause the crash (Never) is applied correctly to the ATX2AT Smart Convert? 
(If it doesn't sleep after 4+ hours)

**Comment by SzieberthAdam** [comment]
Sure, I will do it tomorrow.

**Comment by SzieberthAdam** [comment]
It is applied.

Note: I finally tried the ATX2AT with a 0dB SilverStone PSU. It is amazing I have a silent 486! Thanks for the unit!

**Comment by x86fr** [comment]
Ok. I will correct that bug asap.

**Comment by x86fr** [comment]
Solved with 0.4b
